arches/app/permissions/arches_permission_base.py

```diff
@@ -503,32 +503,35 @@ def user_can_read_concepts(self, user: User) -> bool:
         """
 
         if user.is_authenticated:
-            return user.groups.filter(name="RDM Administrator").exists()
+            return self.user_in_group_by_name(user, ["RDM Administrator"])
         return False
 
     def user_is_resource_editor(self, user: User) -> bool:
         """
         Single test for whether a user is in the Resource Editor group
         """
 
-        return user.groups.filter(name="Resource Editor").exists()
+        return self.user_in_group_by_name(user, ["Resource Editor"])
 
     def user_is_resource_reviewer(self, user: User) -> bool:
         """
         Single test for whether a user is in the Resource Reviewer group
         """
 
-        return user.groups.filter(name="Resource Reviewer").exists()
+        return self.user_in_group_by_name(user, ["Resource Reviewer"])
 
     def user_is_resource_exporter(self, user: User) -> bool:
         """
         Single test for whether a user is in the Resource Exporter group
         """
 
-        return user.groups.filter(name="Resource Exporter").exists()
+        return self.user_in_group_by_name(user, ["Resource Exporter"])
 
     def user_in_group_by_name(self, user: User, names: Iterable[str]) -> bool:
-        return bool(user.groups.filter(name__in=names))
+        for group in user.groups.all():
+            if group.name in names:
+                return True
+        return False
```

Comment:
This populates the cache (if not prefetched already) so that we avoid going to the db every time to test this.